configuration.go

```diff
@@ -107,6 +116,10 @@ func (config *Configuration) validate() error {
 		config.Endpoint = defaultEndpoint
 	}
 
+	if config.ServiceName == "" {
+		config.ServiceName = "unknown_service"
```

Comment:
If our library doesn't set service name, do the default resource attributes set it to "unknown_service"? If so, I suggest we remove any default on our side and allow the default to come from OTel in case of future changes.

Reply:
I read some OTEL docs and saw line "SDKs should fall back to unknown_service name in case none is provided" but you are right, we can rely on default resource :)